src/libraries/System.Net.Mail/tests/Functional/MockSmtpServer.cs

```diff
@@ -0,0 +1,265 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Net;
+using System.Net.Mail;
+using System.Net.Sockets;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Systen.Net.Mail.Tests
+{
+    public class MockSmtpServer : IDisposable
```

Comment:
nit: We tend to use the word 'Fake' instead of 'Mock'.  See [examples](https://github.com/dotnet/runtime/tree/master/src/libraries/System.Net.Http.WinHttpHandler/tests/UnitTests).

Strictly speaking mocks and stubs are both types of fakes. Historically, we have designed System.Net networking tests using principles from "The Art of Unit Testing" book (which I recommend).

Here is a good definition I like between "mock", "stub", and "fake":
https://stackoverflow.com/questions/3459287/whats-the-difference-between-a-mock-stub

>A fake is a generic term that can be used to describe either a stub or a mock object (handwritten or otherwise), because they both look like the real object. Whether a fake is a stub or a mock depends on how it’s used in the current test. If it’s used to check an interaction (asserted against), it’s a mock object. Otherwise, it’s a stub.

So, in that context, the 'SmtpServer' here is really a 'stub' since you are not asserting against it specifically but simply using it as part of the tests. So, it should be called 'StubSmtpServer' because it is really a 'stub' and not a 'mock'. 

So, I would suggest 'StubSmtpServer' or just 'FakeSmtpServer' but not 'MockSmtpServer'.

Other opinions on this topic are welcome. :)

Reply:
Sure I can rename it, I don't have a strong preference here